proxy_wasm_intrinsics_full.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2016-2019 Envoy Project Authors
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * API Available to WASM modules.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+#define PROXY_WASM_PROTOBUF_FULL 1
```

Comment:
This is defined twice, once in `copts` and in once the header file.

Reply:
Yes, yes I did.   This is unfortunately AFAICT it is required because bazel doesn't seem to work the way one would reasonably expect it to work.  The copts doesn't seem to be contagious to targets which depend on a a header from the library which defines the copt.  Consequently in order to get the #define to actually impact the header when it is included in another target, ,e.g. gcpc_call_cpp.cc in test/extensions/filters/http/wasm/test_data, on needs to add it to that target or conversely one could use a header which includes it such as here.

That said, my understanding of bazel is perhaps not what it should be and if you have a better solution I am all ears.

Reply:
Yeah, `copts` are not propagated "upwards", so having this defined in headers is fine... With that said, is there any reason to have it defined in `copts` as well?

Reply:
I don't want to add the copts in the wasm_cc_binary target because that is messy.  This way I can conceal that mess in the proxy-wasm-cpp-sdk library proper.

Reply:
You don't need copts in either place, just having this header should be enough.

Reply:
Erm, I meant that you should remove define from `copts` in `:proxy_wasm_intrinsics_lite` and `: proxy_wasm_intrinsics_full` targets.

Reply:
Then I have to add it at the higher level in the wasm_cc_binary targets which is messy and inconvenient.  I can't include a different header in the proxy_wasm_intrinsics.cc file without having a #define so I need two targets for it with different copts.   If you have a way to get this to work I would like to here it but I tried several ways and this was the best way I could find to get bazel to do what I want.  I suppose I could build wasm_cc_binary_full and wasm_cc_binary_lite, but that  doesn't seem to be an improvement either.

Reply:
`wasm_cc_binary_full` and `wasm_cc_binary_lite` sound messy, let's not do that.

I still think that `copts` is not necessary, but it's also not a blocker, so feel free to merge this as-is.